cpp/oneapi/dal/backend/primitives/objective_function/logloss_dpc.cpp

```diff
@@ -610,6 +639,15 @@ sycl::event logloss_hessian_product<Float>::compute_without_fit_intercept(
         last_iter_deps = { update_grad_e };
     }
 
+    if (comm_.get_rank_count() > 1) {
+        sycl::event::wait_and_throw(last_iter_deps);
+        {
+            ONEDAL_PROFILER_TASK(hessp_allreduce);
+            auto hessp_arr = dal::array<Float>::wrap(q_, out.get_mutable_data(), out.get_count());
```

Comment:
I believe you [can pass event list](https://github.com/oneapi-src/oneDAL/blob/14ad1ababe5601a62983c49b0ac2e046c990c445/cpp/oneapi/dal/array.hpp#L201) as an arument to the wrap function.